# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/rental/tests/map.rs
# Commit: 16350e08d34f
# Full Hash: 16350e08d34ff06c7f46745c6b160c7a7bda9140
# Author: Dana Keeler <dkeeler@mozilla.com>
# Date: 2019-12-06 04:43:57
# Regressor Bug: 1586855
# File Overlap Count: 6
# Description:
#   bug 1586855 - incorporate CRLite filters into cert_storage r=jcj,kjacobs
#   
#   This patch implements CRLite lookups for TLS server certificate revocation
#   information in telemetry-only mode. It adds a new preference
#   "security.pki.crlite_mode" to control the behavior of this feature. Setting
# ==============================================================================

diff -r 64fd40663930 -r 16350e08d34f third_party/rust/rental/tests/map.rs
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/third_party/rust/rental/tests/map.rs	Thu Dec 05 22:41:53 2019 +0000
@@ -0,0 +1,42 @@
+#[macro_use]
+extern crate rental;
+
+
+pub struct Foo {
+	i: i32,
+}
+
+
+rental! {
+	mod rentals {
+		use super::*;
+
+		#[rental(map_suffix = "T")]
+		pub struct SimpleRef<T: 'static> {
+			foo: Box<Foo>,
+			fr: &'foo T,
+		}
+
+		#[rental_mut(map_suffix = "T")]
+		pub struct SimpleMut<T: 'static> {
+			foo: Box<Foo>,
+			fr: &'foo mut T,
+		}
+	}
+}
+
+
+#[test]
+fn map() {
+	let foo = Foo { i: 5 };
+	let sr = rentals::SimpleRef::new(Box::new(foo), |foo| foo);
+	let sr = sr.map(|fr| &fr.i);
+	assert_eq!(sr.rent(|ir| **ir), 5);
+
+	let foo = Foo { i: 12 };
+	let sm = rentals::SimpleMut::new(Box::new(foo), |foo| foo);
+	let sm = sm.map(|fr| &mut fr.i);
+	assert_eq!(sm.rent(|ir| **ir), 12);
+}
+
+